MC/bin/o2dpg_sim_workflow.py

```diff
@@ -627,38 +630,46 @@ def getDPL_global_options(bigshm=False, ccdbbackend=True):
    SGN_CONFIG_task=createTask(name='gensgnconf_'+str(tf), tf=tf, cwd=timeframeworkdir)
    SGN_CONFIG_task['cmd'] = 'echo "placeholder / dummy task"'
    if GENERATOR == 'pythia8':
-      SGN_CONFIG_task['cmd'] = '${O2DPG_ROOT}/MC/config/common/pythia8/utils/mkpy8cfg.py \
-                                --output=pythia8.cfg                                     \
-                                --seed='+str(TFSEED)+'                                   \
-                                --idA='+str(PDGA)+'                                      \
-                                --idB='+str(PDGB)+'                                      \
-                                --eCM='+str(ECMS)+'                                      \
-                                --eA='+str(EBEAMA)+'                                     \
-                                --eB='+str(EBEAMB)+'                                     \
-                                --process='+str(PROCESS)+'                               \
-                                --ptHatMin='+str(PTHATMIN)+'                             \
-                                --ptHatMax='+str(PTHATMAX)
-      if WEIGHTPOW   > 0:
-         SGN_CONFIG_task['cmd'] = SGN_CONFIG_task['cmd'] + ' --weightPow=' + str(WEIGHTPOW)
+      # see if config is given externally
+      externalPythia8Config = simInitialConfigKeys.get("GeneratorPythia8", {}).get("config", None)
+      if externalPythia8Config != None:
+         print ("Using external Pythia8 configuration " + str(externalPythia8Config))
+         # in this case, we copy the external config to the local dir (maybe not even necessary)
+         SGN_CONFIG_task['cmd'] = 'cp ' + externalPythia8Config + ' pythia8.cfg'
```

Comment:
This will be executed inside the TF directory. So if one passes a relative path to the sim WF creation, the original file might not be found from inside the TF directory, right?

Reply:
yes ... we should always use absolute/canonical paths for everything in O2DPG. Maybe need to add a special check function for this purpose.

Reply:
one could do
```python
externalPythia8Config = abspath(externalPythia8Config)
if not exists(externalPythia8Config):
    # abort...
# now it can be used anywhere because we have an absolute path now
```